controllers/rack.go

```diff
@@ -86,14 +86,28 @@ func (r *SingleClusterReconciler) reconcileRacks() reconcileResult {
 
 		failedPods, _ := getFailedAndActivePods(podList)
 		if len(failedPods) != 0 {
-			r.Log.Info("Upgrading or rolling restart rack with failed pods", "rackID", state.Rack.ID, "failedPods", failedPods)
+			r.Log.Info("Reconcile the failed pods in the Rack", "rackID", state.Rack.ID, "failedPods", failedPods)
 
-			_, res = r.upgradeOrRollingRestartRack(found, state, ignorablePods, failedPods)
-			if !res.isSuccess {
+			if res = r.reconcileRack(
+				found, state, ignorablePods, failedPods,
+			); !res.isSuccess {
+				return res
+			}
+
+			r.Log.Info("Reconciled the failed pods in the Rack", "rackID", state.Rack.ID, "failedPods", failedPods)
+		}
+
+		// Use old podList as if there are new failed pods then they will not recover with this rolling restart
+		failedPods, _ = getFailedAndActivePods(podList)
```

Comment:
Here we are using old podList, can there be a case where due to scale down happend above, pod got deleted but podList will have the object of same.

Reply:
After every scaledown we requeue the request, so I don't think that above scenario will occur

Reply:
got it, then there is no reason of fetching `failedPods` again here as `podlist` is same and `getFailedAndActivePods` has all static checks on the list.

Reply:
This is for scenario CR is updated and pods go in pending/failed state and then again CR is updated to revert the changes. In this case spec and status are same and above flow won't handle failed pod. This condition is added separately outside to handle scale-up even with failed pods.

Reply:
yes I know, here my point is about calling `failedPods, _ = getFailedAndActivePods(podList)`. Is it ever gonna give you different failedPods than above line 87?

Reply:
rolling restart should recover failed pods